modules/library/render/src/main/java/org/geotools/renderer/label/LabelCacheImpl.java

```diff
@@ -1211,7 +1211,9 @@ private void setupPointTransform(
                 (textStyle.getAnchorY() * (textBounds.getHeight()))
                         - textStyle.getDisplacementY()
                         - textBounds.getHeight()
-                        + painter.getLineHeight();
+                        + ((painter.lines.size() == 1)
```

Comment:
I believe you found the right area to fix, but the change works fine for y anchor equals to 0, which makes for top alignment. But what if bottom alignment was asked and the last line was significantly smaller than the first one instead?
Try the following, change the y anchor to 1, and the string to "Espace\nabcd ef", and the result is this (with your code changes in):

![selezione_577](https://user-images.githubusercontent.com/325433/53584720-82576c00-3b84-11e9-9d31-0c584ee52e40.png)

Before the code change it would have been instead:

![image](https://user-images.githubusercontent.com/325433/53584828-d3676000-3b84-11e9-8f76-341c23f354ea.png)


I believe that the code should be sensitive to the anchor value, at least for the cases where a human visible alignment is evident:

- if anchorY is 1 (or very close, use a tolerance), use first line height
- if anchorY is 0  (or very close, use a tolerance), use the last line height
- for any other value, I'd use the average line height instead, just to care for the 0.5 anchor case better, the others are hardly verifiable

Reply:
understood